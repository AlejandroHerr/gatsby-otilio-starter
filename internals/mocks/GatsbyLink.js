/* eslint-disable */
import React from 'react';

const GatsbyLink = ({ children, className, to }) => (
  <a className={className} href={to}>
    {children}
  </a>
);

export default GatsbyLink;
