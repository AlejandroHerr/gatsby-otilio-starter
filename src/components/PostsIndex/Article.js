import React from 'react';
import Link from 'gatsby-link';

import { articleType } from '../../propTypes/postsIndex';

import styles from './PostsIndex.module.scss';

const getDisplayDate = date => date.toLocaleDateString('en-us', { day: 'numeric', month: 'short', year: 'numeric' });

const Article = ({ article }) => {
  const {
    author,
    date,
    excerpt,
    slug,
    tags,
    title,
  } = article;

  const postDate = new Date(date);

  return (
    <article className={styles.posts_index_article}>
      <div className={styles.posts_index_article__inner}>
        <header className="post-header">
          <h2 className={styles.posts_index_article__title}>
            <Link
              className={styles.posts_index_article__title__link}
              to={slug}
            >
              {title
              }
            </Link>
          </h2>
          <span className={styles.posts_index_article__meta}>
            {author}
            {tags && ` on ${tags.join(', ')}`}
            {' | '}
            <time dateTime={postDate.toISOString()}>{getDisplayDate(postDate)}</time>
          </span>
          <div className={styles.posts_index_article__clear} />
        </header>
        <section>
          <p className={styles.posts_index_article__excerpt}>{excerpt}</p>
        </section>
      </div>
      <div className={styles.posts_index_article__separator} />
    </article>
  );
};

Article.propTypes = {
  article: articleType.isRequired,
};

export default Article;
