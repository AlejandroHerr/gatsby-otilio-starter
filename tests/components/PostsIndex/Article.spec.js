import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GatsbyLink from 'gatsby-link';

import Article from '../../../src/components/PostsIndex/Article';

const defaultProps = {
  article: {
    author: 'testAuthor',
    date: '2018-04-30T05:00:00.000Z',
    excerpt: 'Excerpt test',
    slug: 'test',
    title: 'testTitle',
  },
};

const setup = (props = {}) => {
  const article = shallow(<Article {...defaultProps} {...props} />);

  return {
    article,
    link: article.find(GatsbyLink),
  };
};

describe('PostsIndex', () => {
  describe('Article', () => {
    it('should render an article link', () => {
      const { article, link } = setup();

      expect(link).toHaveLength(1);
      expect(link.first().prop('to')).toBe(defaultProps.article.slug);
      expect(article).toMatchSnapshot();
    });
    it('should render an article link with tags', () => {
      const props = {
        article: {
          ...defaultProps.article,
          tags: ['test1', 'test2'],
        },
      };
      const tree = renderer.create(<Article {...props} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
