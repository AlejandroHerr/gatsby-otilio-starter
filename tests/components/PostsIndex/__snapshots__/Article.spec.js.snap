// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsIndex Article should render an article link 1`] = `
<article
  className="posts_index_article"
>
  <div
    className="posts_index_article__inner"
  >
    <header
      className="post-header"
    >
      <h2
        className="posts_index_article__title"
      >
        <GatsbyLink
          className="posts_index_article__title__link"
          to="test"
        >
          testTitle
        </GatsbyLink>
      </h2>
      <span
        className="posts_index_article__meta"
      >
        testAuthor
         | 
        <time
          dateTime="2018-04-30T05:00:00.000Z"
        >
          Apr 30, 2018
        </time>
      </span>
      <div
        className="posts_index_article__clear"
      />
    </header>
    <section>
      <p
        className="posts_index_article__excerpt"
      >
        Excerpt test
      </p>
    </section>
  </div>
  <div
    className="posts_index_article__separator"
  />
</article>
`;

exports[`PostsIndex Article should render an article link with tags 1`] = `
<article
  className="posts_index_article"
>
  <div
    className="posts_index_article__inner"
  >
    <header
      className="post-header"
    >
      <h2
        className="posts_index_article__title"
      >
        <a
          className="posts_index_article__title__link"
          href="test"
        >
          testTitle
        </a>
      </h2>
      <span
        className="posts_index_article__meta"
      >
        testAuthor
         on test1, test2
         | 
        <time
          dateTime="2018-04-30T05:00:00.000Z"
        >
          Apr 30, 2018
        </time>
      </span>
      <div
        className="posts_index_article__clear"
      />
    </header>
    <section>
      <p
        className="posts_index_article__excerpt"
      >
        Excerpt test
      </p>
    </section>
  </div>
  <div
    className="posts_index_article__separator"
  />
</article>
`;
